### PR TITLE
Documented VB conversion optimization

### LINKED ISSUE #8073: Document floating-point to Integer optimization
## Document floating-point to Integer optimization

Visual Basic 15.8 includes an optimization that requires documentation. The performance of floating-point-to-Integer conversion in the CInt(Fix(*floating-point*)) method calls has been improved and needs to be emphasized.

### PR DESCRIPTION
**Pull request by rpetrusha**
## Documented VB conversion optimization

Fixes #8073 

//cc @KathleenDollard

### CODE REVIEW

**Comment by BillWagner** [comment]
Overall, this is great @rpetrusha 

I'll do a formal review once you're ready.

**Review by BillWagner — COMMENTED** [review]
I read this through, and I had a few initial comments.

**Comment by KathleenDollard** [comment]
Looks good

File: docs/visual-basic/language-reference/functions/type-conversion-functions.md [hunk 1]
```diff
@@ -121,7 +121,7 @@ CUShort(expression)
...
 |`CDec`|[Decimal Data Type](../../../visual-basic/language-reference/data-types/decimal-data-type.md)|+/-79,228,162,514,264,337,593,543,950,335 for zero-scaled numbers, that is, numbers with no decimal places. For numbers with 28 decimal places, the range is +/-7.9228162514264337593543950335. The smallest possible non-zero number is 0.0000000000000000000000000001 (+/-1E-28).|  
-|`CInt`|[Integer Data Type](../../../visual-basic/language-reference/data-types/integer-data-type.md)|-2,147,483,648 through 2,147,483,647; fractional parts are rounded.<sup>1</sup>|  
+|`CInt`|[Integer Data Type](../../../visual-basic/language-reference/data-types/integer-data-type.md)|-2,147,483,648 through 2,147,483,647; fractional parts are rounded.<sup>1</sup> <br/><br/>Starting with Visual Basic 15.8, Visual Basic optimizes the performance of floating-point to integer conversion. To take advantage of this optimization, you pass the value returned by the <xref:Microsoft.VisualBasic.Conversion.Fix%2A> function to the `CInt` function. See the [CInt Example](#cint-example) section for an example. |
```
[the remark this answers is not in the document]
Comment (rpetrusha) [reply docs/visual-basic/language-reference/functions/type-conversion-functions.md]: Great suggestion. Thanks, @BillWagner

**Comment by rpetrusha** [comment]
This is ready for a final review, @BillWagner @KathleenDollard @DualBrain @gafter

**Review by BillWagner — APPROVED** [review]
This LGTM @rpetrusha 

There is still one outstanding comment from @mairaw and then you can :shipit:

**Pull request merged**